### PR TITLE
[Docs] language server pages

### PR DESCRIPTION
**Pull request by predragnikolic**
added:
- suggestion for installing LSP-json
- added instructions for installing plugins like LSP-angular, LSP-graphql and other plugins that were missing.
- discord icon in the footer
- ST4 and ST3 Client Configuration tabs

changed:
- point to LSP-* instantiation instructions instead of manual configuration
- renamed "Server Configuration" to "Client Configuration"

removed:
- removed deprecated sourcegraph's go-language server (in favour of gopls)
- remove polymer editor service
- remove intellij configuration option (in favour of Eclipse JDT Language Server)
- remove microsoft's python language server option (in favour of pyright and pyls)
- removed lua-lsp reference (in favour of lua-ls client configuration)
- removed SBT, DOTTY scala config references (in favour of LSP-metals)

Should I bring any of the removed configurations back?

I think the best way to see the difference is to start the development server locally, and compare it to the current site https://sublimelsp.github.io/LSP/language_servers/

----

Note: I noticed that some links are broken (for [example](https://sublimelsp.github.io/LSP/language_servers/#cc) the `C/C++ guide` link). That is due to the `/lsp` prefix that gh pages adds for repos. I added `site_url: https://sublimelsp.github.io/LSP/` to the `mkdocs.yaml` file, but I am not sure if that is gonna fix it. 

Maybe I will have to include the lsp prefix in the links:
```diff
- [C/C++ guide](/guides/cplusplus)
+ [C/C++ guide](/lsp/guides/cplusplus)
```
If you know a solution for the link problem, please let me know :)

### CODE REVIEW

**Comment by jwortmann** [comment]
Looks like all relative links are broken. Could you try
```diff
- [list of language servers](/language_servers/)
+ [list of language servers](language_servers.md)
```
in index.md and
```diff
- [C/C++ guide](/guides/cplusplus)
+ [C/C++ guide](guides/cplusplus.md)
```

https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages

**Comment by predragnikolic** [comment]
Thanks @jwortmann :)

**Comment by rchl** [comment]
Would it make sense and work for `mkdocs.yml` to live in the `docs` folder?

File: docs/language_servers.md [hunk 1]
```diff
@@ -1,307 +1,428 @@
...
-Please remember to put the configurations in a `"clients"` dictionary in your `LSP.sublime-settings` file, as shown in the example above.
-If you use or would like to use language servers that are not in this list, please create issues or pull requests, so we can add support for more languages.
+Follow the setup steps for a language server to get it up and running.
 
-### Bash<a name="bash"></a>
+If you have problems see the [common issues](troubleshooting.md#common-problems) page or search the [LSP/issues](https://github.com/sublimelsp/LSP/issues) before opening new ones.
 
-1. Make sure you have `node` installed.
-2. Download the [LSP-bash](https://packagecontrol.io/packages/LSP-bash) helper package.
+If there are no setup steps for a language server on this page, but a [language server implementation](https://microsoft.github.io/language-server-protocol/implementors/servers/) exist, follow the guide for [creating a client configuration](/guides/client_configuration/). Pull requests for adding a new client configuration are welcome.
```
[the remark this answers is not in the document]
Comment (predragnikolic) [reply docs/language_servers.md]: Thanks Rafal,
resolved with bcfb6b4

**Comment by predragnikolic** [comment]
@rchl It makes sense that docs related files are in the docs folder, 
so looked up on how to move the mkdocs file and I stumbled upon this:

https://github.com/mkdocs/mkdocs/issues/1276

I am not that bothered with the `mkdocs.yml` file being at the root. 
But If you are, there is one solution suggested by one of the maintainers of mkdocs:
https://github.com/mkdocs/mkdocs/issues/1276#issuecomment-327921779 (the `docs/src` solution seems ok)

**Comment by rchl** [comment]
> @rchl It makes sense that docs related files are in the docs folder,
> so looked up on how to move the mkdocs file and I stumbled upon this:
> 
> [mkdocs/mkdocs#1276](https://github.com/mkdocs/mkdocs/issues/1276)
> 
> I am not that bothered with the `mkdocs.yml` file being at the root.
> But If you are, there is one solution suggested by one of the maintainers of mkdocs:
> [mkdocs/mkdocs#1276 (comment)](https://github.com/mkdocs/mkdocs/issues/1276#issuecomment-327921779) (the `docs/src` solution seems ok)

From that comment it seems possible if we create `/docs/src/` directory and put `mkdocs.yml` in `/docs/`. I don't want to make the call though. We'll see what @rwols has to say.

BTW. The `/site/` directory should be git-ignored.

**Comment by rwols** [comment]
Putting things in docs/src sounds good to me!

**Review by rwols — APPROVED** [review]
Looks good to me :)

**Pull request merged**